src/edu/uw/easysrl/syntax/parser/ParserBuilder.java

```diff
@@ -30,24 +30,12 @@
 public abstract class ParserBuilder<T extends ParserBuilder<T>> {
 
 	ParserBuilder(final File modelFolder) {
-		this.modelFolder = modelFolder;
-		this.jointModel = new File(modelFolder, "weights").exists();
-		try {
-			this.unaryRules = AbstractParser.loadUnaryRules(new File(modelFolder, "unaryRules"));
-			this.lexicalCategories = TaggerEmbeddings.loadCategories(new File(modelFolder, "categories"));
-			this.seenRules = new SeenRules(new File(modelFolder, "seenRules"), lexicalCategories);
-			final File cutoffsFile = new File(modelFolder, "cutoffs");
-			cutoffs = cutoffsFile.exists() ? Util.deserialize(cutoffsFile) : null;
-			if (new File(modelFolder, "markedup").exists()) {
-				Coindexation.parseMarkedUpFile(new File(modelFolder, "markedup"));
-			}
-		} catch (final IOException e) {
-			throw new UncheckedIOException(e);
-		}
-
+		modelFolder(modelFolder);
 	}
 
-	final File modelFolder;
+	ParserBuilder() { }
```

Comment:
I didn't have this constructor before, because I thought the model folder should be a mandatory argument. Is there a time you wouldn't want to specify it?